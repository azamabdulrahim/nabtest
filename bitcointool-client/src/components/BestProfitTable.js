import React, { Component } from "react";

class BestProfitTable extends Component {
  render() {
    const { data } = this.props;

    let display;
    if (data) {
      display = (
        <table className="table table-bordered">
          <tbody>
            <tr>
              <th colSpan="2">{data.date}</th>
            </tr>
            <tr className="table-active">
              <td colSpan="2">{data.currency}</td>
            </tr>
            <tr>
              <td>Buy</td>
              <td>Sell</td>
            </tr>
            <tr>
              <td>${data.buy.price}</td>
              <td>${data.sell.price}</td>
            </tr>
            <tr>
              <td>{data.buy.time}</td>
              <td>{data.sell.time}</td>
            </tr>
            <tr>
              <td colSpan="2">Profit: ${data.profit}</td>
            </tr>
          </tbody>
        </table>
      );
    }

    return <div>{display}</div>;
  }
}

export default BestProfitTable;
