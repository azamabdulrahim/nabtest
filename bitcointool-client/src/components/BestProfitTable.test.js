import React from "react";
import { shallow } from "enzyme";
import BestProfitTable from "./BestProfitTable";

describe("BestProfitTable", () => {
  let wrapper;
  it("renders correctly with no data", () => {
    wrapper = shallow(<BestProfitTable />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("table").length).toEqual(0);
  });

  it("renders correctly with data", () => {
    const data = {
      date: "07/05/2018",
      currency: "BTC",
      buy: {
        time: "09:30 AM",
        price: "2.30"
      },
      sell: {
        time: "04:30 PM",
        price: "2.58"
      },
      profit: "0.28"
    };

    wrapper = shallow(<BestProfitTable data={data} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.containsMatchingElement(<span>No data</span>)).toBeFalsy();
    expect(wrapper.find("table").length).toEqual(1);
    expect(wrapper.html().search("07/05/2018")).toBeGreaterThan(0);
    expect(wrapper.html().search("BTC")).toBeGreaterThan(0);
    expect(wrapper.html().search("09:30 AM")).toBeGreaterThan(0);
    expect(wrapper.html().search("04:30 PM")).toBeGreaterThan(0);
    expect(wrapper.html().search("2.30")).toBeGreaterThan(0);
    expect(wrapper.html().search("2.58")).toBeGreaterThan(0);
    expect(wrapper.html().search("0.28")).toBeGreaterThan(0);
  });
});
