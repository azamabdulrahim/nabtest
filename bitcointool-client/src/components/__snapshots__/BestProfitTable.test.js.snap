// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestProfitTable renders correctly with data 1`] = `
<div>
  <table
    className="table table-bordered"
  >
    <tbody>
      <tr>
        <th
          colSpan="2"
        >
          07/05/2018
        </th>
      </tr>
      <tr
        className="table-active"
      >
        <td
          colSpan="2"
        >
          BTC
        </td>
      </tr>
      <tr>
        <td>
          Buy
        </td>
        <td>
          Sell
        </td>
      </tr>
      <tr>
        <td>
          $
          2.30
        </td>
        <td>
          $
          2.58
        </td>
      </tr>
      <tr>
        <td>
          09:30 AM
        </td>
        <td>
          04:30 PM
        </td>
      </tr>
      <tr>
        <td
          colSpan="2"
        >
          Profit: $
          0.28
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`BestProfitTable renders correctly with no data 1`] = `<div />`;
